core/src/main/java/org/springframework/security/access/expression/ExpressionUtils.java

```diff
@@ -19,12 +19,42 @@
 import org.springframework.expression.EvaluationContext;
 import org.springframework.expression.EvaluationException;
 import org.springframework.expression.Expression;
+import org.springframework.security.authorization.AuthorizationDecision;
+import org.springframework.security.authorization.ExpressionAuthorizationDecision;
 
 public final class ExpressionUtils {
 
 	private ExpressionUtils() {
 	}
 
+	/**
+	 * Evaluate a SpEL expression and coerce into an {@link AuthorizationDecision}
+	 * @param expr a SpEL expression
+	 * @param ctx an {@link EvaluationContext}
+	 * @return the resulting {@link AuthorizationDecision}
+	 * @since 6.3
+	 */
+	public static AuthorizationDecision evaluate(Expression expr, EvaluationContext ctx) {
+		try {
+			Object result = expr.getValue(ctx);
+			if (result instanceof AuthorizationDecision decision) {
+				return decision;
+			}
+			if (result instanceof Boolean granted) {
+				return new ExpressionAuthorizationDecision(granted, expr);
+			}
+			if (result == null) {
+				return null;
+			}
```

Comment:
Have you considered using `AuthorizationDecision decision = expr.getValue(ctx, AuthorizationDecision.class)` and let the `ConversionService` do the work?

For example, you can change the `DefaultMethodSecurityExpressionHandler#createEvaluationContext` to include a custom `TypeConverter`, something like:

```java
GenericConversionService conversionService = new GenericConversionService();
conversionService.addConverter(new BooleanToAuthorizationDecisionConverter());
StandardTypeConverter typeConverter = new StandardTypeConverter(conversionService);
ctx.setTypeConverter(typeConverter);
```

Reply:
I suppose I don't see the value as the logic is quite simple. Further, the conversion service wouldn't have access to the `Expression` so it couldn't construct an `ExpressionAuthorizationDecision`, which is needed for backward compatibility.